spec/src/main/asciidoc/jsonb.asciidoc

```diff
@@ -0,0 +1,148 @@
+
+.JsonbProperty Java Code Example
+[source,java,numbered]
+----
+public class Widget {
+    @JsonbProperty("widgetName")
+    private String name;
+    private double weight;
+    private int quantity;
+    //...
+
+    @JsonbProperty("shippingWeight")
+    public double getWeight() {
+        return weight;
+    }
+    //...
+
+    @JsonbProperty("qty")
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+}
+----
+
+That would create a schema like this:
+
+.JsonbProperty Schema Example
+[source,graphql,numbered]
+----
+type Widget @_mappedType(type : "__internal__") {
+  widgetName: String @_mappedOperation(operation : "__internal__")
+  quantity: Int! @_mappedOperation(operation : "__internal__")
+  shippingWeight: Float! @_mappedOperation(operation : "__internal__")
+}
+
+input WidgetInput @_mappedType(type : "__internal__") {
+  widgetName: String @_mappedInputField(inputField : "__internal__")
+  qty: Int! @_mappedInputField(inputField : "__internal__")
+  weight: Float! @_mappedInputField(inputField : "__internal__")
+}
+----
+
+
+Similarly, the `@JsonbTransient` annotation can be used to ignore certain fields from the type or input type in the 
+schame. The same rules apply: if the annotation is on the getter, then the field is ignored in the type; if the
+annotation is on the setter, then the field is ingored in the input type; if the annotation is on the Java field, it
+is ignored in both.
+
+Here is an example, similar to the previous example:
+
+.JsonbTransient Java Code Example
+[source,java,numbered]
+----
+public class Widget {
+    @JsonbTransient
+    private String name;
+    private double weight;
+    private int quantity;
+    //...
+
+    @JsonbTransient
+    public double getWeight() {
+        return weight;
+    }
+    //...
+
+    @JsonbTransient
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+}
+----
+
+That would create a schema like this:
+
+.JsonbTransient Schema Example
+[source,graphql,numbered]
+----
+type Widget @_mappedType(type : "__internal__") {
+  quantity: Int! @_mappedOperation(operation : "__internal__")
+}
+
+input WidgetInput @_mappedType(type : "__internal__") {
+  weight: Float! @_mappedInputField(inputField : "__internal__")
+}
+----
+
+=== Runtime Data Transformation
+
+JSON-B annotations like `@JsonbDateFormat` and `@JsonbNumberFormat` can be used to transform data at runtime.
+
+For example, suppose you have a Java entity like this:
+
+.JsonbDateFormat Java Code Example
+[source,java,numbered]
+----
+public class Widget {
+    @JsonbDateFormat("yyyy-MM-dd")
+    private LocalDate constructionDate;
+}
+----
+
+A query that requests a Widget's construction date would get a result like this:
+
+.JsonbTransient Schema Example
+[source,json,numbered]
+----
+{
+    "constructionDate": "2019-09-16"
+}
+----
+
+Note that this information is not automatically added to the schema, so it is considered a best practice to document
```

Comment:
I did not know that ? I thought it's part of the schema? Can it not be ?

Reply:
It might be possible to add it as a description comment.  Maybe the behavior should be to include the date or number format as the description if there is no description provided by the user?

Something like:
```
public class EnrollmentPeriod {

    @JsonbDateFormat("yyyy-MM-dd")
    private LocalDate startDate;

    @JsonbDateFormat("yyyy-MM-dd")
    @Description("The last day to enroll")
    private LocalDate endDate;
}
```

would generate:

```
input EnrollmentPeriodInput @_mappedType(type : "__internal__") {
  # Date Format: yyyy-MM-dd
  startDate: String @_mappedInputField(inputField : "__internal__")
  # The last day to enroll
  endDate: String @_mappedInputField(inputField : "__internal__")
}
```

Another alternative might be to append the date/number format to the description provided by the user (if any).  

What do you think?  Maybe we should open another issue to address documenting date/number formats in the schema?

Reply:
Should we discuss 8n the next meeting ? I would like to get Bojan's view

Reply:
Consensus at today's meeting was that yes, we should try to document the date format unless the user provides their own description (in which case, they are advised to also document the format).

Additionally, Bojan suggested that we should use custom scalars for Date, Time, and DateTime - these should essentially just be Strings, but by using custom scalars it is clear to the client that the format is important.

Reply:
With changes on the way using `@Description` in issue #44, I think it makes more sense to add the date format (and number format) to the description in the schema under a separate issue and PR. I will open a new issue for it shortly and reference this conversation.